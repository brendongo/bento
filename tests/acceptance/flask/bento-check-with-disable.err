Running Bento checks...


⚠ 12 finding(s) in  s
◦ To suppress all findings, run $ bento archive.
