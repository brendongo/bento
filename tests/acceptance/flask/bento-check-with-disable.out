==> Bento Summary
warning: r2c.bandit.B307 https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b307-eval
     --> src/flask/cli.py:892:0
      |
  892 |            with open(startup, "r") as f:
  893 |                eval(compile(f.read(), startup, "exec"), ctx)
      |
      = note: Use of possibly insecure function - consider using safer ast.literal_eval.                                                                                                                              


advice: r2c.bandit.B110 https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html
     --> tests/test_appctx.py:74:0
      |
   74 |            raise Exception("dummy")
   75 |        except Exception:
   76 |            pass
      |
      = note: Try, Except, Pass detected.                                                                                                                                                                             

advice: r2c.bandit.B110 https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html
     --> tests/test_appctx.py:93:0
      |
   93 |                raise Exception("dummy")
   94 |            except Exception:
   95 |                pass
      |
      = note: Try, Except, Pass detected.                                                                                                                                                                             


advice: r2c.bandit.B110 https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html
     --> tests/test_basic.py:784:0
      |
  784 |                raise TypeError()
  785 |            except Exception:
  786 |                pass
      |
      = note: Try, Except, Pass detected.                                                                                                                                                                             

advice: r2c.bandit.B110 https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html
     --> tests/test_basic.py:796:0
      |
  796 |                raise TypeError()
  797 |            except Exception:
  798 |                pass
      |
      = note: Try, Except, Pass detected.                                                                                                                                                                             

error: r2c.bandit.B201 https://bandit.readthedocs.io/en/latest/plugins/b201_flask_debug_true.html
     --> tests/test_basic.py:1912:0
      |
 1912 |        hostname, port = "localhost", 8000
 1913 |        app.run(hostname, port, debug=True)
 1914 |        assert rv["result"] == "running on %s:%s ..." % (hostname, port)
      |
      = note: A Flask app appears to be run with debug=True, which exposes the Werkzeug debugger and allows the execution of arbitrary code.                                                                          

error: r2c.flask.secure-set-cookie https://checks.bento.dev/en/latest/flake8-flask/secure-set-cookie
     --> tests/test_basic.py:1956:9
      |
 1956 |           r.set_cookie("foo", "bar" * 100)
      |
      = note: Flask cookies should be handled securely by setting secure=True, httponly=True, and samesite='Lax' in set_cookie(...).  If your situation calls for different settings, explicitly disable the settin...


advice: r2c.bandit.B110 https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html
     --> tests/test_reqctx.py:45:0
      |
   45 |            raise Exception("dummy")
   46 |        except Exception:
   47 |            pass
      |
      = note: Try, Except, Pass detected.                                                                                                                                                                             

advice: r2c.bandit.B110 https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html
     --> tests/test_reqctx.py:64:0
      |
   64 |                raise Exception("dummy")
   65 |            except Exception:
   66 |                pass
      |
      = note: Try, Except, Pass detected.                                                                                                                                                                             


error: r2c.bandit.B201 https://bandit.readthedocs.io/en/latest/plugins/b201_flask_debug_true.html
     --> tests/test_templating.py:404:0
      |
  404 |        app.run(debug=True)
  405 |        assert app.templates_auto_reload
      |
      = note: A Flask app appears to be run with debug=True, which exposes the Werkzeug debugger and allows the execution of arbitrary code.                                                                          


r2c.bandit:
  B110                  6▕▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬
  B201                  2▕▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬
  B307                  1▕▬▬▬▬▬▬▬▬▬

r2c.flask:
  secure-set-cookie     1▕▬▬▬▬▬▬▬▬▬

